core/storage/btree.rs

```diff
@@ -2431,14 +2443,71 @@ impl BTreeCursor {
                     &parent_page,
                     balance_info,
                     parent_contents,
-                    pages_to_balance_new,
+                    &pages_to_balance_new,
                     page_type,
                     leaf_data,
                     cells_debug,
                     sibling_count_new,
                     rightmost_pointer,
```

Comment:
Validation should be after balance shallower. Can we add validation for balance shallower too?